### PR TITLE
Serve metrics docs

### PR DESCRIPTION
**Pull request by architkulkarni**
This PR adds basic documentation for the metrics exposed by Serve.  It also prefixes the serve metrics with `serve_` for easier searching.  Some typos are fixed and the help string formatting has been made more consistent.

Basic documentation is included for creating custom metrics for Serve.  This PR adds methods `serve.get_current_backend_tag()` and `serve.get_current_backend_tag()` that can be called from the user's backend code.  These are used to get the tags to use for custom metrics defined in the user's backend code.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/api.py [hunk 1]
```diff
@@ -617,6 +626,40 @@ def connect() -> Client:
...
+    Raises:
+        RayServeException if not called from within a Ray Serve backend.
+    """
+    if _INTERNAL_REPLICA_CONTEXT is None:
+        raise RayServeException("Called `serve.get_current_backend_tag()`"
+                                "from a function that is not running in a "
+                                "Ray Serve backend on this cluster, so it "
+                                "has no backend tag yet. Please "
+                                "use client.create_backend() to run your "
+                                "backend function using Ray Serve.")
```
Comment (edoakes) [inline python/ray/serve/api.py:641]: I think we could make this more concise & clearer by just saying `serve.get_current_backend_tag() may only be called from within a Ray Serve backend.`

**Review by edoakes — APPROVED** [review]
Nice, this looks great! My only concern is if we should be calling them "tags," but that's what it has been called throughout the API so looks fine for now.

We should also modify the existing metrics in backend_worker.py to use these APIs instead of whatever bespoke plumbing they're using now.

File: python/ray/serve/api.py (hunk 1 above)
Comment (edoakes) [reply python/ray/serve/api.py:641]: (same for replica tag)

File: python/ray/serve/api.py [hunk 2]
```diff
@@ -617,6 +626,40 @@ def connect() -> Client:
...
+    else:
+        return _INTERNAL_REPLICA_CONTEXT.backend_tag
+
+
+def get_current_replica_tag() -> ReplicaTag:
+    """When called from within a backend, return its replica tag.
```
Comment (edoakes) [inline python/ray/serve/api.py:647]: "Replica tag" is not currently a user-facing concept, so we need to explain it a bit here. Can you add a description? e.g., "uniquely identifies a single replica (process) for a backend"

**Comment by walli55** [comment]
How long is the one batter

**Comment by walli55** [comment]
Until tomorrow when you come back and

**Comment by walli55** [comment]
#14458

**Pull request merged**